scripts/move_patterns.py

```diff
@@ -0,0 +1,268 @@
+       [-0.60, -0.10, -1.571, "Move: Leg K" ], \
+       [-0.40, -0.30, -0.785, "Move: Leg L" ], \
+       [-0.20, -0.30,  0.000, "Move: Leg M" ], \
+       [ 0.00, -0.10,  0.785, "Move: Leg N" ], \
+       [ 0.00,  0.10,  1.571, "Move: Leg O" ], \
+       [ 0.20,  0.30,  0.785, "Move: Leg P" ]  \
+       ]
+
+
+    #######################################################################################
+
+    """
+    Constructor for our class
+    """
+    def __init__(self):
+
+       rospy.init_node('controller')
+
+       # Time per loop for the main control
+       self.loop_msec = 50
+
+       # Define waitAtEachVertex as 0 for continual moves or 1 for pause each vertex
+       waitAtEachVertex = 1
+
+       waypointName = 'line'
+       waypointList = self.figureLine
+       scaleX  = 1.0
+       scaleY  = 1.0
+       offsetX = 0.0
+       offsetY = 0.0
+
+       # read commandline arguments and place them in array
+       # Only the -w option seems to work, something is wrong with getopt usage here
+       try:
+           opts, args = getopt.getopt(sys.argv[1:], 'hcw:s:x:y:h', \
+               ['help','continue', 'waypoints=','scale=','offsetX=','offsetY='])
+       except getopt.GetoptError as err:
+           # will print something like "option -a not recognized"
+           print "Error in recognized options"  
+           printUsage()
+           sys.exit(2)
+
+       for o, a in opts:
+           # evaluate given options
+           if o in ("-h", "--help"):
+               print ("displaying help")
+               printUsage()
+               sys.exit(2)
+           elif o in ("-c", "--continue"):
+               # continuous operation, do not require ENTER at each vertex
+               waitAtEachVertex = 0
+           elif o in ("-w", "--waypoints"):
+               # define a waypoint list for waypoints
+               waypointName = a
+               if (a == "line"):
+                   waypointList = self.figureLine
+               elif (a == "box"):
+                   waypointList = self.figureBox
+               elif (a == "figure8"):
+                   waypointList = self.figure8octagon
+               elif (a == "octagon"):
+                   waypointList = self.figureOctagon
+               else:
+                   print ("Invalid choice of waypoint list")
+                   sys.exit(2)
+               print "Waypoint list will be '%s'", waypointList
+           elif o in ("-s", "--scaleX"):
+               scaleX = float(a)
+               scaleY = float(a)
+           elif o in ("-x", "--offsetX"):
+               offsetX = float(a)
+           elif o in ("-y", "--offsetY"):
+               offsetY = float(a)
+
+       print ("WaypointsName: %s scaleX %f scaleY %f offsetX %f offsetY %f" \
+             %(waypointName,scaleX,scaleY,offsetX,offsetY))
+
+       print "A total of %d waypoints is in the list " % (len(waypointList))
+
+       # continue going through waypoints over and over.
```

Comment:
Lines above this belong in the constructor, since they are setting up the state, but I think lines below this should be in `run()`, since they are doing the work.  This would make your class more reusable for other use cases.

Reply:
Yes, that would be cleaner.  It will be done.